zipkin-collector/core/src/main/java/zipkin2/collector/Collector.java

```diff
@@ -56,14 +51,35 @@ public static Builder newBuilder(Class<?> loggingClass) {
 
   public static final class Builder {
     final Logger logger;
+    final ArrayList<CollectedSpanHandler> collectedSpanHandlers = new ArrayList<>();
     StorageComponent storage;
-    CollectorSampler sampler;
     CollectorMetrics metrics;
 
     Builder(Logger logger) {
       this.logger = logger;
     }
 
+    /** @see {@link CollectorComponent.Builder#sampler(CollectorSampler)} */
+    public Builder sampler(CollectorSampler sampler) {
+      if (sampler == null) throw new NullPointerException("sampler == null");
+      this.collectedSpanHandlers.add(0, sampler); // sample first
+      return this;
+    }
+
+    /**
+     * @see {@link CollectorComponent.Builder#addCollectedSpanHandler(CollectedSpanHandler)}
+     * @since 2.17
+     */
+    public Builder addCollectedSpanHandler(CollectedSpanHandler collectedSpanHandler) {
+      if (collectedSpanHandler == null) {
+        throw new NullPointerException("collectedSpanHandler == null");
+      }
+      if (collectedSpanHandler != CollectedSpanHandler.NOOP) { // lenient on config bug
+        this.collectedSpanHandlers.add(collectedSpanHandler);
```

Comment:
I really hate the idea of duplicates, but there was a change in brave where @anuraaga felt strongly about letting duplicates exist. I can move the "this will cause bugs" disclaimer here also unless rag can hopefully change his mind https://github.com/openzipkin/brave/pull/961

Reply:
my mistake, he revised his opinion. I will revert the other change, but drop silently.

Reply:
https://github.com/openzipkin/brave/pull/1001